operators/node/src/main.rs

```diff
@@ -0,0 +1,269 @@
+use actix_web::client::Client;
+use kube::api::{Api, DeleteParams, ListParams, Meta, PostParams};
+use kube_derive::CustomResource;
+use log::info;
+use mbus_api::*;
+use serde::{Deserialize, Serialize};
+use std::convert::TryFrom;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+struct CliArgs {
+    /// The Rest Server URL to connect to
+    /// Default: 127.0.0.1:8080
+    #[structopt(long, short, default_value = "localhost:8080")]
+    rest: String,
+
+    /// REST Path to get the nodes
+    #[structopt(long, short, default_value = "/nodes")]
+    nodes: String,
+
+    /// Polling period
+    #[structopt(long, short, default_value = "30s")]
+    period: humantime::Duration,
+}
+
+#[derive(CustomResource, Deserialize, Serialize, Clone, Debug)]
+#[kube(
+    group = "openebs.io",
+    version = "v1alpha1",
+    kind = "MayastorNode",
+    namespaced
+)]
+#[kube(apiextensions = "v1beta1")]
+#[kube(status = "String")]
+#[serde(rename_all = "camelCase")]
+pub struct MayastorNodeSpec {
+    pub grpc_endpoint: String,
+}
+
+impl TryFrom<&MayastorNode> for Node {
+    type Error = strum::ParseError;
+    fn try_from(kube_node: &MayastorNode) -> Result<Self, Self::Error> {
+        Ok(Node {
+            registration: Register {
+                id: kube_node.name(),
+                grpc_endpoint: kube_node.spec.grpc_endpoint.clone(),
+            },
+            state: kube_node
+                .status
+                .as_ref()
+                .unwrap_or(&"".to_string())
+                .parse()?,
+        })
+    }
+}
+
+#[actix_web::main]
+async fn main() -> anyhow::Result<()> {
+    env_logger::init_from_env(
+        env_logger::Env::default()
+            .filter_or(env_logger::DEFAULT_FILTER_ENV, "info"),
+    );
+
+    let polling_period = CliArgs::from_args().period.into();
+    let rest_client = Client::default();
+
+    let rest_url = format!("http://{}", CliArgs::from_args().rest);
+    let rest_path = CliArgs::from_args().nodes;
+    let rest_uri = format!("{}{}", rest_url, rest_path);
+    info!("Using REST URI: {}", rest_uri);
+
+    let kube_client = kube::Client::try_default().await?;
+    let namespace = "mayastor";
+
+    // Validate that our "CRD" is up to date?
+
+    // Manage the MayastorNode CR
+    let nodes_api: Api<MayastorNode> =
+        Api::namespaced(kube_client.clone(), namespace);
+
+    loop {
+        // Poll for kubernetes nodes and rest nodes
+        // Reconcile from rest into kubernetes
+        if let Err(error) =
+            polling_work(&nodes_api, &rest_client, &rest_uri).await
```

Comment:
Do we ensure that k8s API server is not burdened with too many requests?

Reply:
No, it only sleeps 30s (or whatever we decide) between each poll.